src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -493,17 +493,39 @@ internal static string TupleMemberName(int position)
             return "Item" + position;
         }
 
+        private static bool IsElementNameForbidden(string name)
+        {
+            switch (name)
+            {
+                case "CombineHashCodes":
```

Comment:
Yes, I'd put only the public members that are guaranteed to be there. - ToString, Rest, Create...
We are going to do the member lookup in most cases anyways and that will catch if something else is public and has a clash.